Objects/unionobject.c

```diff
@@ -0,0 +1,473 @@
+// types.Union -- used to represent e.g. Union[int, str], int | str
+#include "Python.h"
+#include "pycore_object.h"
+#include "structmember.h"
+
+
+typedef struct {
+    PyObject_HEAD
+    PyObject *args;
+} unionobject;
+
+static void
+unionobject_dealloc(PyObject *self)
+{
+    unionobject *alias = (unionobject *)self;
+
+    _PyObject_GC_UNTRACK(self);
+    Py_XDECREF(alias->args);
+    self->ob_type->tp_free(self);
+}
+
+static Py_hash_t
+union_hash(PyObject *self)
+{
+    unionobject *alias = (unionobject *)self;
+    Py_hash_t h1 = PyObject_Hash(alias->args);
+    if (h1 == -1) {
+        return -1;
+    }
+    return h1;
+}
+
+static int
+union_traverse(PyObject *self, visitproc visit, void *arg)
+{
+    unionobject *alias = (unionobject *)self;
+    Py_VISIT(alias->args);
+    return 0;
+}
+
+static PyMemberDef union_members[] = {
+    {"__args__", T_OBJECT, offsetof(unionobject, args), READONLY},
+    {0}
+};
+
+static PyObject *
+union_getattro(PyObject *self, PyObject *name)
```

Comment:
You can remove this function and place `PyObject_GenericGetAttr` directly on the slot:

```
    .tp_getattro = PyObject_GenericGetAttr,

```